libpromises/generic_agent.c

```diff
@@ -1968,3 +1968,100 @@ void SetupSignalsForAgent(void)
     signal(SIGUSR1, HandleSignalsForAgent);
     signal(SIGUSR2, HandleSignalsForAgent);
 }
+
+void ShowContextsFormatted(EvalContext *ctx)
```

Comment:
Since it only gets an `EvalContext` argument and it operates only on it, I would say it belongs to `eval_context.c`.